mantidproject/mantid
# D7 YIG bank offset correction

Corrected an issue of inconsistent sign when setting offset to detector banks and propagating that information to the bank offset saved in the calibration output. Also, expanded the system test to include testing the absolute positions coming out of calibration for all three wavelengths and two fitting methods (individual and global).


**To test:**

1. Using mantidpython or from the workbench, run calibration using the following definition:

```
Load('ILL/D7/396442_396831.nxs', OutputWorkspace='intermediateWavelengthScan')
approximate_wavelength = '4.8' # Angstrom
D7YIGPositionCalibration(InputWorkspace='intermediateWavelengthScan', ApproximateWavelength=approximate_wavelength,
 			                  YIGPeaksFile='D7_YIG_peaks.xml', CalibrationOutputFile='test_intermediateWavelength.xml',
			                  MinimalDistanceBetweenPeaks=1.5, BankOffsets=[3, 3, -1],
			                  MaskedBinsRange=[-50, -25, 15],
			                  FittingMethod='Individual', FitOutputWorkspace='intermediateWavelength')
```

2. In the workbench, load some D7 data, e.g. 401800.nxs from the UnitTest data, and specify the `PositionCalibration` property to `YIGFile`, and provide the position calibration XML as the `YIGFilename` 
3. Open the instrument viewer and observe whether the gaps between bank2 (detector ID 44) and bank3 (detector ID 45), and  bank3 (detector ID 88) and bank4 (ID 89) are about the same in magnitude and there are no overlapping detectors.

*There is no associated issue.*

*This does not require release notes* because it fixes an internal issue.




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
